resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -234,20 +218,26 @@ private[spark] class ApplicationMaster(args: ApplicationMasterArguments) extends
     resources.toMap
   }
 
-  def getAttemptId(): ApplicationAttemptId = {
-    client.getAttemptId()
+  def getAttemptId(sparkConf: SparkConf): ApplicationAttemptId = {
+    client.getAttemptId(sparkConf)
   }
 
   final def run(): Int = {
     doAsUser {
-      runImpl()
+      runImpl(
+        if (isClusterMode) {
+          runDriver()
+        } else {
+          runExecutorLauncher()
+        }
+      )
     }
     exitCode
   }
 
-  private def runImpl(): Unit = {
+  private def runImpl(opBlock: => Unit): Unit = {
```

Comment:
There are things in this method that don't look right when you think about an unmanaged AM.

e.g., overriding `spark.master`, `spark.ui.port`, etc, look wrong.

The handling of app attempts also seems wrong, since with an unmanaged AM you don't have multiple attempts. Even the shutdown hooks seems a bit out of place.

Seems to me it would be easier not to try to use this method for the unmanaged AM.